### ISSUE TITLE
ARM Support for docker image

### ISSUE THREAD

**Issue by daxroc** [grafana/rollout-operator#16]
Current rollout image does not have an arm64 compatible image

```
standard_init_linux.go:228: exec user process caused: exec format error
```

I just moved the build to the dockerfile using the builder pattern and use the builtin support for multi arch from `docker buidlx` 
```
FROM --platform=$BUILDPLATFORM golang:1.17-alpine AS builder

ADD . /app
WORKDIR /app

ARG TARGETOS
ARG TARGETARCH

RUN CGO_ENABLED=0 GOOS=${TARGETOS} GOARCH=${TARGETARCH} \
    go build -ldflags="-s -w" ./cmd/rollout-operator

FROM alpine:3.14

COPY --from=builder /app/rollout-operator /bin/rollout-operator
ENTRYPOINT [ "/bin/rollout-operator" ]

# Create rollout-operator user to run as non-root.
RUN addgroup -g 10000 -S rollout-operator && \
    adduser  -u 10000 -S rollout-operator -G rollout-operator
USER rollout-operator:rollout-operator

ARG revision
LABEL org.opencontainers.image.title="rollout-operator" \
      org.opencontainers.image.source="https://github.com/grafana/rollout-operator" \
      org.opencontainers.image.revision="${revision}"
```

And to build just specify the architectures (Note arm builds require push ~ images aren't stored locally)

```
docker buildx build --platform linux/amd64,linux/arm64 --tag your-registry.tld/repo/grafana/rollout-operator:latest --push .
```

**Comment by tobernguyen** [comment]
Thanks @daxroc 

I modified the `Dockerfile` a bit so it can work with the latest version (`v0.2.0`)

```
FROM golang:1.18-alpine AS builder

ADD . /app
WORKDIR /app

ARG TARGETOS
ARG TARGETARCH

RUN CGO_ENABLED=0 GOOS=${TARGETOS} GOARCH=${TARGETARCH} \
    go build -ldflags='-extldflags "-static"' ./cmd/rollout-operator

FROM alpine:3.16

COPY --from=builder /app/rollout-operator /bin/rollout-operator
ENTRYPOINT [ "/bin/rollout-operator" ]

# Create rollout-operator user to run as non-root.
RUN addgroup -g 10000 -S rollout-operator && \
    adduser  -u 10000 -S rollout-operator -G rollout-operator
USER rollout-operator:rollout-operator

ARG revision
LABEL org.opencontainers.image.title="rollout-operator" \
      org.opencontainers.image.source="https://github.com/grafana/rollout-operator" \
      org.opencontainers.image.revision="${revision}"

```

Building & pushing command is the same

```
docker buildx build --platform linux/amd64,linux/arm64 --tag your-registry.tld/repo/grafana/rollout-operator:latest --push .
```

**Comment by isac322** [comment]
https://hub.docker.com/r/isac322/grafana-rollout-operator
I created multi arch image that follows upstream via dependabot

**Comment by rhysxevans** [comment]
Hi

Are there any timelines on getting an officially supported ARM (ideally multiarch) image created and added to the helm chart ?

Thanks

**Comment by rhysxevans** [comment]
Thank you

**Comment by andyasp** [comment]
Hey all, thanks for the suggestions and highlighting this issue. The newest release (https://github.com/grafana/rollout-operator/releases/tag/v0.4.0) has been published and the image associated with it supports both `linux/amd64` and `linux/arm64`.

> Are there any timelines on getting an officially supported ARM (ideally multiarch) image created and added to the helm chart ?

If you're referring to the rollout-operator helm chart that has now been updated. I've also opened a PR for Mimir's chart, but there's more moving parts there so it's harder to say when that will ultimately be released.

> Thank you

Cheers!
